### DISCUSSION TITLE
adding two mesh generators

### DISCUSSION THREAD

**Discussion by starkeyk** [idaholab/moose#18320] [Q&A Meshing] [answered]
I am trying to create periodic boundary conditions on the displacement field and constrain one of the corner nodes to have zero displacement (to get rid of rigid body translations). I was able to do this in an older version of MOOSE (before mesh generators), but now I get an error 

"Your MeshGenerator tree contains multiple possible generator outputs :
"sidesets and one or more of the following from an independent set: "ns1"

This may be due to a missing dependency or may be intentional. Please select the final MeshGenerator in
the [Mesh] block with the "final_generator" parameter or add additional dependencies to remove the ambiguity."

The way that I created the boundary side sets is using the SideSetsFromNormalsGenerator and to create the corner node I use ExtraNodesetGenerator with coord='0 0 0'. I think this error is referring to the fact that the side sets contain the extra node set that I am creating. For some reason, this wasn't a problem in the older code (I think the Dirichlet BC had precedence over the periodic BC's) but it is now.
 Is there another way to implement these BC's?

**Comment by GiudGiud** [comment, marked as answer]
Hello 

Could you please paste your [Mesh] / mesh generators block here?

Guillaume

**Reply by starkeyk** [reply]
[Mesh]
  [./msh]
    type = FileMeshGenerator
    file = ../../test.exo
    dim = 3
  []
  [./sidesets]
    type = SideSetsFromNormalsGenerator
    input = msh
    normals = '0  0  1
               0  0 -1
               0  1  0
               0 -1  0
               1  0  0
               -1 0  0'
    fixed_normal = true
    new_boundary = 'top1 bottom1 front1 back1 right1 left1'
  []
  [./ns1]
    type = ExtraNodesetGenerator
    input = msh
    new_boundary = 'nodeset1'
    coords = '0 0 0'
  [../]
[]

**Reply by GiudGiud** [reply]
so the problem is that `ns1` should take `sidesets` as input, not `msh`

Right now there are two final meshes, like two branches. MOOSE cant choose, so it asks for a final_generator parameter. But the fix is to create a single mesh incorporating both changes

**Reply by starkeyk** [reply]
when write sidesets as input to the node set generator I get the following error:

The following error occurred in the object "ns1", of type "ExtraNodesetGenerator".

Node set can not be empty!

**Reply by GiudGiud** [reply]
what about inverting the chain?
so msh -> ns1 -> sidesets 

Also is there a node at 0,0,0 in test.exo?

**Reply by starkeyk** [reply]
there was not a node located at 0,0,0. I think it is working now with the following input configuration:

[Mesh]
  [./msh]
    type = FileMeshGenerator
    file = ../../test.exo
    dim = 3
  []
   [./sidesets]
    type = SideSetsFromNormalsGenerator
    input = msh
    normals = '0  0  1
               0  0 -1
               0  1  0
               0 -1  0
               1  0  0
               -1 0  0'
    fixed_normal = true
    new_boundary = 'top1 bottom1 front1 back1 right1 left1'
  []
  [./ns1]
    type = ExtraNodesetGenerator
    input = sidesets
    new_boundary = 'nodeset1'
    nodes = '1'
  [../]
[]
